### ISSUE TITLE
bug: fails to process error responses. 

### ISSUE THREAD

**Issue by ayzu** [AfterShip/aftership-sdk-go#34]
Currently, the API client fails to process an error response from the create tracking endpoint.

**Steps to reproduce:**

```
package main

import (
	"context"
	"fmt"
	"time"

	"github.com/aftership/aftership-sdk-go/v2"
)

func main() {
	unauthorizedTracker, err := aftership.NewClient(aftership.Config{APIKey: "UNAUTHORIZED"})
	if err != nil {
		panic(err)
	}

	_, err = unauthorizedTracker.CreateTracking(context.Background(),
		aftership.CreateTrackingParams{Slug: []string{"total-express"}, TrackingNumber: time.Now().String()})

	fmt.Println(err)
}
```

**Expected behaviour**

The function call should return an error of type [APIError](https://github.com/AfterShip/aftership-sdk-go/blob/f26cbe5d76accbc514db2ab3e4ce45b74fed5ad7/errors.go#L17) that contains the error code and error message ("unauthorized").

**Actual behaviour**

Returns the unstructured error:

_error unmarshalling the JSON response: json: cannot unmarshal array into Go struct field Tracking.data.tracking.slug of type string_

As far as I understand, the problem arises from line 78 in requests.go file, where we try to unmarshal the response. The expected response has field `Tracking.Slug` of type string, while the received type is a slice of strings.

**Comment by sunznx** [comment]
fix at #35

**Comment by Bossa573** [comment]
Hello @ayzu,
thank you for the reporting and would be fixed by https://github.com/AfterShip/aftership-sdk-go/pull/35 and v2.0.2
